### ISSUE TITLE
Support for Apple M1

### ISSUE THREAD

**Issue by leinad1** [ibmdb/python-ibmdb#688]
Is there a way to support Apple M1?
I am on Monterey 12.0.1.

I am facing the following error:
ImportError: dlopen(/opt/homebrew/Caskroom/miniforge/base/lib/python3.9/site-packages/ibm_db.cpython-39-darwin.so, 0x0002): symbol not found in flat namespace '_SQLAllocHandle'

**Comment by amukherjee28** [comment]
Could you try installing the ibm_db driver using the --no-cache-dir option.

pip install ibm_db --no-cache-dir and let me know if that works.

Unfortunately we dont have a test system with the above OS, so cannot check this for now.

**Comment by leinad1** [comment]
uninstalled and installed with option above, but no change.
In the meantime, I found out, that it is not working only with homebrewed python installation.
"Normal" python installation does work: /usr/local/bin/python3

**Comment by klaasbrant** [comment]
The issue is that is no Db2 client for ARM processors. That has to be fixed before the python client can be fixed (if it needs fixing at all).

**Comment by bimalkjha** [comment]
@klaasbrant The newer clidriver does work on M1 chip processor. Few user has tested. If you see above post by leinad1, problem is only with homebrewed python. Thanks.

**Comment by klaasbrant** [comment]
@bimalkjha are you sure? When you look at the files in https://public.dhe.ibm.com/ibmdl/export/pub/software/data/db2/drivers/odbc_cli/ there is only 1 macos file which is x86
I just tried to install on rasberry pi (also ARM like Mac M1) and it fails because it cannot resolve the platform

**Comment by bimalkjha** [comment]
@klaasbrant If you see [this comment](https://github.com/ibmdb/node-ibm_db/issues/784#issuecomment-894249100) by one of the user of M1 Chip system, db2level and db2cli commands works fine on Mac M1 system. So, I updated here. One other github user christo-ph also confirmed it.
I am not sure about Rasberry pi or Alpine Linux yet, but it seems the macos64_clidriver.tar.gz works for both x64 and arm64 processor. I am yet to get access of an M1 system, so I have not tested it. Once, I'll verify then will update here. Thanks.

**Comment by jonathanherdt** [comment]
This issue is not solved for me, still getting the error mentioned by OP.

I investigated a bit. I downloaded the package from [PyPi](https://pypi.org/project/ibm-db/#files) and ran the installation with `setup.py install`, using first the system Python version (located in `/usr/bin/python3`, for me Python 3.8.9) and then the Python version from my virtual environment, Python 3.8.10. Both come to the same result:

```
> /usr/bin/python3 setup.py install
[...]
clang -bundle -undefined dynamic_lookup -arch arm64 -arch x86_64 -Wl,-headerpad,0x1000 -L/opt/homebrew/opt/openssl@3/lib -I/opt/homebrew/opt/openssl@3/include build/temp.macosx-10.14.6-arm64-3.8/ibm_db.o -Lclidriver/lib -L$ORIGIN/clidriver/lib -ldb2 -o build/lib.macosx-10.14.6-arm64-3.8/ibm_db.cpython-38-darwin.so
ld: warning: directory not found for option '-L$ORIGIN/clidriver/lib'
ld: warning: ignoring file clidriver/lib/libdb2.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
```

Using the files that result from this build ends up in the same error as experienced by regularly installing the library, the one described by OP:

```
    import ibm_db
E   ImportError: dlopen(.../venv/lib/python3.8/site-packages/ibm_db.cpython-38-darwin.so, 0x0002): symbol not found in flat namespace '_SQLAllocHandle'
```

After reading through everything, I also tried to use a manually downloaded `clidriver` (from [here](https://public.dhe.ibm.com/ibmdl/export/pub/software/data/db2/drivers/odbc_cli/), this does not change the result.

**Comment by vinaydeep0711** [comment]
I am trying to install odbc , pymysql and mysql connector on the my mac os M1 air machine and getting the same error 
symbol not found in flat namespace '_SQLAllocHandle-
    return __import__("pyodbc")
ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/pyodbc.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace '_SQLAllocHandle'

Can someone help me to resolve this.

**Comment by bimalkjha** [comment]
@vinaydeep0711 This forum is only for issues related to `ibm_db` and not for `pyodbc`. Please post your problem in different forum. Thanks.

**Comment by bimalkjha** [comment]
@vinaydeep0711 This forum is only for issues related to `ibm_db` and not for `pyodbc`. Please post your problem in different forum. Thanks.

**Comment by bimalkjha** [comment]
@leinad1 @klaasbrant We have released `ibm_db v3.2.5` with native installation support for MacOS Silicon Chip system and no need to use the emulated environment with x86_64 python on M1 chip system. Please install latest version of ibm_db and verify. Thanks.
